src/colorizer/ColorizeCanvas.ts

```diff
@@ -178,12 +185,36 @@ export default class ColorizeCanvas {
     this.colorMapRangeMax = 0;
     this.currentFrame = 0;
 
+    this.overlay = new CanvasOverlay();
+    this.showScaleBar = false;
+    this.frameToCanvasScale = new Vector4(1, 1, 1, 1);
+
     this.render = this.render.bind(this);
     this.getCurrentFrame = this.getCurrentFrame.bind(this);
     this.setOutOfRangeDrawMode = this.setOutOfRangeDrawMode.bind(this);
+    this.updateScaling = this.updateScaling.bind(this);
   }
 
-  get domElement(): HTMLCanvasElement {
+  /**
+   * The DOM element containing the canvas and its overlay.
+   */
+  get domElement(): HTMLDivElement {
+    if (!this.canvasContainer) {
+      // Set up the canvas overlay as a sibling in the DOM layout
+      // by creating a dummy parent container.
+      this.canvasContainer = document.createElement("div");
+      this.canvasContainer.appendChild(this.renderer.domElement);
+      this.canvasContainer.appendChild(this.overlay.domElement);
+      this.canvasContainer.style.position = "relative";
+      this.overlay.domElement.style.position = "absolute";
+      this.overlay.domElement.style.left = "0";
+      this.overlay.domElement.style.top = "0";
+    }
+
+    return this.canvasContainer;
+  }
```

Comment:
Doing all this element-building in the getter could open up surprising results later. E.g. the client could add `canvasElement` to the document then later added `domElement` in a different place, thereby lifting the canvas out of its original context. I'd suggest putting this in the constructor? If it's important to have the option to use either the bare canvas or this wrapper, this could be a method, but I think it makes more sense to let the client treat the canvas as a single drop-in element that contains and abstracts everything.

Reply:
Makes sense, fixed!

Reply:
Makes sense, fixed!